### DISCUSSION TITLE
Discord Rich Presence Support for Ghostty?

### DISCUSSION THREAD

**Discussion by nickpapciak** [ghostty-org/ghostty#3035] [Feature Requests, Ideas] [answered]
It would be cool to have support for discord rich presence like follows...
<img width="289" alt="image" src="https://github.com/user-attachments/assets/3856ec3c-c155-4f1c-914c-b8c0a222b8b3" />

it would require spawning a new thread, and occasionally checking user status, so it would likely impact performance. Is this a feature worth implementing? If so, I think it should be a default-off but can be turned on in the config.

**Comment by mitchellh** [comment, marked as answer]
This sounds like a great thing to ensure our future plugin framework can handle. I'm not sure this should be built-in.

**Reply by jcollie** [reply]
I'd agree that this is prime plugin stuff. A lot of people are probably doing things in their terminal that shouldn't be advertised like that.
